src/toast/Toast.tsx

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import {Icon} from "../icon";
+import {classNames} from "../utils";
+import {ClassNames} from "../modals/ClassNames";
+
+export enum Direction {
+    TOP_RIGHT = "top-right",
+    BOTTOM_LEFT = "bottom-left",
+    BOTTOM_RIGHT = "bottom-right",
+    TOP_LEFT = "top-left",
+}
+
+enum Directions {
+    LEFT = "left",
+    RIGHT = "right",
+    BOTTOM = "bottom",
+    TOP = "top",
+}
+
+export enum MessageType {
+    SUCCESS = "success",
+    WARNING = "warning",
+    DANGER = "danger",
+    INFO = "info",
+}
+
+enum IconType {
+    SUCCESS = "success-standard",
+}
+
+const DEFAULT_HIDE_TIME = 3000;
+
+type ToastProps = {
+    showToast?: boolean;
+    duration?: number;
+    position?: Direction;
+    type?: MessageType;
+    icon?: string;
+    text: string;
+    onClose: Function;
```

Comment:
This can be optional prop. By default toast will close after specified timeout.

Reply:
Addressed comment